### ISSUE TITLE
怎么在自己的数据集上生成dense label map呢？

### ISSUE THREAD

**Issue by wxshan** [zihangJiang/TokenLabeling#7]
怎么在自己的数据集上生成dense label map呢？

**Comment by zihangJiang** [comment]
Hi,
Assume you've got a pre-trained classifier like EfficientNet-B6 for your dataset, you can simply remove the final global average pooling layer (and may have to adjust the shape of the output feature map before you feed it to the classification head) to generate the dense label map.

**Comment by wxshan** [comment]
If I save the feature vector of the picture as pt, how can I make the label of the small image correspond to the label of the whole image later?

**Comment by zihangJiang** [comment]
This is automatically done during training.
https://github.com/zihangJiang/TokenLabeling/blob/aa438eff9b9fc2daa8c8b4cc6bfaa6e3721f995e/tlt/data/mixup.py#L27-L33

The label map will be cropped (according to the random crop box) and resized to the target shape.

**Comment by wxshan** [comment]
Thank you














At 2021-06-18 10:57:01, "蒋子航" ***@***.***> wrote:

This is automatically done during training.
https://github.com/zihangJiang/TokenLabeling/blob/aa438eff9b9fc2daa8c8b4cc6bfaa6e3721f995e/tlt/data/mixup.py#L27-L33

The label map will be cropped (according to the random crop box) and resized to the target shape.

**Comment by zihangJiang** [comment]
You can reopen this issue if you have any further questions.

**Comment by yangzhaopeng3** [comment]
> 

Hi Zihang, thanks for your excellent work.
I'd like to ask how the score map are generated.
First we use a pretrained machine annotator like NFNet-F6, and as you say we remove the final global average pooling layer, we get the feature map of an input image. But how to transform the feature map to the dense score map, let's say 1000*patch_num for ImageNet? Sorry for my unfamiliarity with the NFNet-F6, will it automatically generate the patch label or there are some other technique to finish this?
Thanks for your reply.

**Comment by theneotopia** [comment]
> Hi, Assume you've got a pre-trained classifier like EfficientNet-B6 for your dataset, you can simply remove the final global average pooling layer (and may have to adjust the shape of the output feature map before you feed it to the classification head) to generate the dense label map.

Hi Zihang @zihangJiang , thanks for your excellent work.
I'd like to ask how the score map are generated.
First we use a pretrained machine annotator like NFNet-F6, and as you say we remove the final global average pooling layer, we get the feature map of an input image. But how to transform the feature map to the dense score map, let's say 1000*patch_num for ImageNet? Sorry for my unfamiliarity with the NFNet-F6, will it automatically generate the patch label or there are some other technique to finish this?
Thanks for your reply.
